editoast/fga_migrations/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use fga_migrations::client::ApplyArgs;
+use fga_migrations::client::CliParser;
+use fga_migrations::client::Commands;
+use fga_migrations::client::Migration;
+use fga_migrations::migration::run_migrations;
+
+use fga_migrations::migration::TelemetrySettings;
+use opentelemetry::trace::TracerProvider;
+use opentelemetry_otlp::SpanExporter;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::Resource;
+use opentelemetry_sdk::propagation::TraceContextPropagator;
+use opentelemetry_sdk::resource::EnvResourceDetector;
+use opentelemetry_sdk::resource::SdkProvidedResourceDetector;
+use opentelemetry_sdk::resource::TelemetryResourceDetector;
+use opentelemetry_sdk::trace::SdkTracerProvider;
+use tracing::info;
+use tracing_subscriber::Layer;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+
+use clap::Parser;
+use fga_migrations::migration::Settings;
+
+fn setup_telemetry(telemetry_settings: &TelemetrySettings) {
+    let exporter = SpanExporter::builder()
+        .with_tonic()
+        .with_endpoint(telemetry_settings.endpoint.as_str())
+        .build()
+        .expect("Failed to initialize otlp exporter");
+
+    let resource = Resource::builder()
+        .with_detectors(&[
+            Box::new(SdkProvidedResourceDetector),
+            Box::new(TelemetryResourceDetector),
+            Box::new(EnvResourceDetector::new()),
+        ])
+        .build();
+
+    let provider = SdkTracerProvider::builder()
+        .with_resource(resource)
+        .with_simple_exporter(exporter)
+        .build();
+
+    let otlp_tracer = provider.tracer("fga-migrations");
+
+    let otlp_layer = tracing_opentelemetry::layer()
+        .with_tracer(otlp_tracer)
+        .boxed();
+
+    let env_filter_layer = tracing_subscriber::EnvFilter::builder()
+        .with_default_directive(tracing_subscriber::filter::LevelFilter::INFO.into())
+        .from_env_lossy();
+
+    let fmt_layer = tracing_subscriber::fmt::layer()
+        .pretty()
+        .with_file(true)
+        .with_line_number(true)
+        .boxed();
+
+    tracing_subscriber::registry()
+        .with(otlp_layer)
+        .with(env_filter_layer)
+        .with(fmt_layer)
+        .init();
+
+    info!("Tracing enabled with otlp");
+
+    opentelemetry::global::set_text_map_propagator(TraceContextPropagator::new());
+    opentelemetry::global::set_tracer_provider(provider);
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    run().await
+}
+
+async fn run() -> anyhow::Result<()> {
+    let client = CliParser::parse();
+    let settings: Settings = (&client).into();
```

Comment:
This `into` should be next to `CliParser`, not in the lib (am I missing something?)